src/Microsoft.Health.Fhir.CosmosDb/Features/Search/FhirCosmosSearchService.cs

```diff
@@ -40,10 +40,23 @@ public class FhirCosmosSearchService : SearchService
             SearchOptions searchOptions,
             CancellationToken cancellationToken)
         {
-            return await ExecuteSearchAsync(
+            SearchResult searchResult = await ExecuteSearchAsync(
                 _queryBuilder.BuildSqlQuerySpec(searchOptions),
                 searchOptions,
                 cancellationToken);
+
+            if (searchOptions.IncludeTotal == TotalType.Accurate && !searchOptions.CountOnly)
```

Comment:
Are we intentionally computing the total count every time search is executed? Alternatively, we could just compute the total during the first search and somehow carry that information through ct or snapshot or something so we don't need to compute it again and again right? (although we will need to figure out how to deal with the fact that resources could be manipulated while the user is paging through the search result).

Reply:
Right - I've chatted with @johnstairs about this. I think this is an optimization that is definitely worth considering.

I created [#AB70827](https://microsofthealth.visualstudio.com/Health/_workitems/edit/70827).